ocrd/ocrd/workspace.py

```diff
@@ -205,6 +205,8 @@ def add_file(self, file_grp, content=None, **kwargs):
             file_grp,
             kwargs.get('local_filename'),
             content is not None)
+        if 'pageId' not in kwargs:
+            raise ValueError("workspace.add_file must be passed a 'pageId' kwarg, even if it is None.")
```

Comment:
Yes, that's an elegant solution. There is no in between page-associated or global annotations, and processors must not be allowed to forget deciding which is right.